File: LongestCommonPrefix/longestCommonPrefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
  let minLength = strs[0].length;
  strs.forEach(e => {
    minLength = minLength < e.length ? minLength : e.length;
  });
  let longestPrefix = "";
  longestPrefix = lookForLongestPrefix(strs, 0, longestPrefix, minLength);
  return longestPrefix;
};


function lookForLongestPrefix(string, i, longestPrefix, minLength) {
  let stop = false;
  let last_string = "";

  if (i >= minLength) {
    return longestPrefix;
  }

  string.forEach(e => {
    last_string = last_string == "" ? e[i] : last_string;
    if (e[i] != last_string) {
      stop = true;
    }
  })

  if (stop != true) {
    longestPrefix += string[0][i];
    longestPrefix = lookForLongestPrefix(string, i + 1, longestPrefix, minLength)
  }
  return longestPrefix;
}

const string = ["flower", "flow", "flight", "fls"];

console.log(longestCommonPrefix(string));

File: LongestCommonPrefix/longestCommonPrefix.py
class Solution():
    def longestCommonPrefix(self, strs):
        minVal = min(strs, key=len)
        longest = ""
        for i in range(len(minVal)):
            current = strs[0][i]
            for items in strs:
                if current != items[i]:
                    return longest
            longest += current
        
        return longest

strs = ["flower", "flow", "flight"]
sol = Solution()
print(sol.longestCommonPrefix(strs))